### DISCUSSION TITLE
This MICROSOFT AUTOGEN was built and made for Mac ONLY ... Nothing works on Windows.  SIGALRM is not even compatible with WINDOWS.

### DISCUSSION THREAD

**Discussion by videofeedback** [microsoft/autogen#1277] [General]
This MICROSOFT AUTOGEN was built and made for Mac ONLY ... Nothing works on Windows.  SIGALRM is not even compatible with WINDOWS.

**Comment by bitnom** [comment]
> This MICROSOFT AUTOGEN was built and made for Mac ONLY ... Nothing works on Windows. SIGALRM is not even compatible with WINDOWS.

It looks like `SIGALRM` is only used when running on posix systems and when not win32. Most of us are probably developing in Linux or WSL, though I'm sure PRs providing better Windows compatibility would be welcomed.

**Comment by ekzhu** [comment]
Thanks for bring it up. Can you raise an issue regarding to the use of SIGALRM is Unix only?

**Comment by afourney** [comment]
A lot of us use WSL, Docker, Codespaces, etc.

**Comment by sonichi** [comment]
> This MICROSOFT AUTOGEN was built and made for Mac ONLY ... Nothing works on Windows. SIGALRM is not even compatible with WINDOWS.

The warning about SIGALRM is outdated. We should remove it. @victordibia @ekzhu can we remove
https://github.com/microsoft/autogen/blob/3608079911f14ac8e9e6c468d004fb71bd4c961f/autogen/code_utils.py#L397-L405
?

**Reply by ekzhu** [reply]
Let's see if we can #1472

**Reply by ekzhu** [reply]
I tried locally and it works for me. But on github action it fails to mount the local directory. 

https://github.com/microsoft/autogen/actions/runs/7719939354/job/21044043831?pr=1472
